dvc/repo/ls.py

```diff
@@ -82,3 +113,22 @@ def _get_target_path_info(repo, target=None):
     if not target:
         return PathInfo(repo.root_dir)
     return PathInfo(repo.root_dir, target)
+
+
+def get_fs_node(path_info, isout=False):
+    from dvc.compat import fspath
+
+    try:
+        isdir = os.path.isdir(fspath(path_info))
```

Comment:
So for now outputs that are directories will be displayed as regular files(not dirs), right? Since they don't exist in the filesystem. Just clarifying.

I guess we could tell if an output is a directory by looking at it's hash, which will end with `.dir` suffix. So we could use `isdir = out.is_dir_checksum() if out.checksum else False` for outs.

Reply:
Not being able to use `out.isdir()` looks confusing, but this is how it is on non-dvc-checked-out working dir. I wonder if we have some bugs related to this.